test/libsolidity/util/TestFileParser.cpp

```diff
@@ -571,8 +571,73 @@ string TestFileParser::Scanner::scanString()
 
 	while (current() != '\"')
 	{
-		str += current();
-		advance();
+		if (current() == '\\')
+		{
+			advance();
+			switch (current())
+			{
+				case '\\':
+					str += current();
+					advance();
+					break;
+				case 'n':
+					str += '\n';
+					advance();
+					break;
+				case 'r':
+					str += '\r';
+					advance();
+					break;
+				case 't':
+					str += '\t';
+					advance();
+					break;
+				case '0':
+					str += '\0';
+					advance();
+					break;
+				case 'x':
+					str += scanHexPart();
+					break;
+				default:
+					str += "\\";
```

Comment:
I think this should either be an error or not result in `\\`, but instead just in `current()`.

Reply:
I agree. An error is more appropriate (less error prone).